cpp/include/raft/core/mdspan.hpp

```diff
@@ -304,24 +304,53 @@ RAFT_INLINE_FUNCTION auto unravel_index(Idx idx,
   }
 }
 
+/**
+ * @brief Const accessor specialization for default_accessor
+ * 
+ * @tparam ElementType 
+ * @param a 
+ * @return std::experimental::default_accessor<std::add_const_t<ElementType>> 
+ */
+template<class ElementType>
+std::experimental::default_accessor<std::add_const_t<ElementType>>
+accessor_of_const(std::experimental::default_accessor<ElementType> a)
+{
+  return {a};
+}
+
+/**
+ * @brief Const accessor specialization for host_device_accessor
+ * 
+ * @tparam ElementType the data type of the mdspan elements
+ * @tparam MemType the type of memory where the elements are stored.
+ * @param a host_device_accessor
+ * @return host_device_accessor<std::experimental::default_accessor<std::add_const_t<ElementType>>, MemType> 
+ */
+template<class ElementType, memory_type MemType>
+host_device_accessor<std::experimental::default_accessor<std::add_const_t<ElementType>>, MemType>
+accessor_of_const(host_device_accessor<std::experimental::default_accessor<ElementType>, MemType> a)
+{
+  return {a};
```

Comment:
This is the correct way to do it, as long as `host_device_accessor` has an element type converting constructor like `default_accessor` does (see [the first constructor here](http://eel.is/c++draft/mdspan.accessor.default.members)).  A straightforward unit test would fail to compile without that constructor.

Reply:
This is a great point, @mhoemmen. It's nice to see the new function being used in the k-means tests but we *should* probably have a dedicated testcase for this (ideally in the mdspan cpp test file).